crates/types/src/config/common.rs

```diff
@@ -806,6 +802,21 @@ pub struct CommonOptionsShadow {
     initialization_timeout: humantime::Duration,
     disable_telemetry: bool,
 
+    metadata_client: MetadataClientOptions,
+    // todo drop in version 1.3
```

Comment:
I feel like we should have a convention for such things that we can easily grep for 😅 `todo(release:1.3.0): ...`?